app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,60 @@
+class PostsController < ApplicationController
+  before_action :get_post, only: [:show, :destroy, :edit, :update]
+  before_action :get_comments, only: [:show]
+  def index
+    @posts = Post.all
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def show
+  end
+
+  def create
+    @post = Post.new(post_params)
+    if @post.save
+      flash[:notice] = 'Post succesfully saved'
+      redirect_to post_path(@post)
+    else
+      flash[:warn] = 'Your post sucks. Please try again'
+      redirect_to new_post_path
+    end
+  end
+
+  def destroy
+    if @post.destroy!
```

Comment:
Don't use the ! version here - it will raise an error if it fails. The regular destroy will return a falsy value if it fails, which is what you want.